### DISCUSSION TITLE
Error in TCPCoilExample?

### DISCUSSION THREAD

**Discussion by JustMe-NL** [eModbus/eModbus#255] [Q&A] [answered]
In TCPCoilExample at function FC_0F there is a sanity check on the number of available databytes based on the number of written coils:

line 109: if (numBytes == (numCoils >> 3) + 1) { ... }

But if the Client writes 8 coils this function fails because it expects 2 bytes while 8 coils fit in 1 byte. 
Am I right in assuming this should be:

if (numBytes == ((numCoils - 1) >> 3) + 1) { ... } 

and 

if (newValue.size() >= (size_t)(((length - 1) >> 3) + 1)) { ... } 

for line 223 in CoilData.cpp?

**Comment by Miq1** [comment, marked as answer]
Should be fixed now in the ``master`` branch. Thank you for pointing me to it!
